src/modeler/AutoRest.Swagger/Model/ValidationUtilities.cs

```diff
@@ -34,10 +41,238 @@ public static bool IsTrackedResource(Schema schema, Dictionary<string, Schema> d
             return false;
         }
 
-        // determine if an operation is xms pageable operation
+        /// <summary>
+        /// Populates a list of 'Resource' models found in the service definition
+        /// </summary>
+        /// <param name="serviceDefinition">serviceDefinition for which to populate the resources</param>
+        /// <returns>List of resource models</returns>
+        public static IEnumerable<string> GetResourceModels(ServiceDefinition serviceDefinition)
+        {
+            if (serviceDefinition.Definitions?.Any() != true)
+            {
+                return new List<string>();
+            }
+
+            var xmsAzureResourceModels = GetXmsAzureResourceModels(serviceDefinition.Definitions);
+
+            // Get all models that are returned by PUT operations (200/201 response)
+            var putOperationsResponseModels = GetOperationResponseModels("put", serviceDefinition).Except(xmsAzureResourceModels);
```

Comment:
now we do `Except(xmsAzureResourceModels)` only on 200 responses of PUTs, but not on 201? Also, note that https://github.com/Azure/autorest/issues/1936#issuecomment-288244278 still doesn't mention that.